From 9c5edc055897c4acc365e8cf874ab2664a68d7ea Mon Sep 17 00:00:00 2001
From: Randy Witt <randy.e.witt@linux.intel.com>
Date: Thu, 10 Sep 2015 11:10:18 -0700
Subject: [PATCH 2/3] zephyr-env.sh: Work with zsh

Without this change ZEPHYR_BASE won't be set up properly when using zsh.
This makes it so the path of the zephyr-env.sh is found sucessfully when
using zsh.

Signed-off-by: Randy Witt <randy.e.witt@linux.intel.com>
---
 zephyr-env.sh | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/zephyr-env.sh b/zephyr-env.sh
index e62d8e2..971be8d 100644
--- a/zephyr-env.sh
+++ b/zephyr-env.sh
@@ -9,7 +9,13 @@ fi
 # run (if it exists) by this script.
 
 # identify OS source tree root directory
-export ZEPHYR_BASE=$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )
+if [ -n "$BASH_SOURCE" ]; then
+    export ZEPHYR_BASE="`dirname $BASH_SOURCE`"
+elif [ -n "$ZSH_NAME" ]; then
+    export ZEPHYR_BASE="`dirname $0`"
+else
+    export ZEPHYR_BASE="`pwd`"
+fi
 
 scripts_path=${ZEPHYR_BASE}/scripts
 echo "${PATH}" | grep -q "${scripts_path}"
-- 
2.4.3

